src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/RestExceptionHandler.java

```diff
@@ -41,6 +41,12 @@
         return toResponseEntity(HttpStatus.BAD_REQUEST, ex.getLocalizedMessage());
     }
 
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<Object> handleIllegalArgumentException(IllegalArgumentException ex) {
+        log.error("Case Id errors: {}", ex.getMessage(), ex);
```

Comment:
Log message should be more generic rather than referring to the first IllegalArgumentException message: e.g. "Illegal Argument Exception:".  

And probably should just be either a `log.warning` or `log.debug` call (@MSancaktutar ??).

Reply:
I would use domain specific custom exception rather than generic `IllegalArgumentException` with 500 status.